knative-operator/pkg/controller/knativekafka/knativekafka_controller.go

```diff
@@ -380,3 +407,33 @@ func (r *ReconcileKnativeKafka) deleteKnativeKafkaSource(instance *operatorv1alp
 	}
 	return nil
 }
+
+func (r *ReconcileKnativeKafka) buildManifest(instance *operatorv1alpha1.KnativeKafka) (*mf.Manifest, error) {
+	combinedManifest := &mf.Manifest{}
+	var err error
+
+	if instance.Spec.Channel.Enabled {
+		combinedManifest, err = mergeManifests(r.rawKafkaChannelManifest.Client, combinedManifest, &r.rawKafkaChannelManifest)
+		if err != nil {
+			return nil, fmt.Errorf("failed to merge KafkaChannel manifest: %w", err)
+		}
+	}
+
+	if instance.Spec.Source.Enabled {
+		combinedManifest, err = mergeManifests(r.rawKafkaSourceManifest.Client, combinedManifest, &r.rawKafkaSourceManifest)
+		if err != nil {
+			return nil, fmt.Errorf("failed to merge KafkaSource manifest: %w", err)
+		}
+	}
+	return combinedManifest, nil
+}
+
+// Merges the given manifests into a new single manifest
+func mergeManifests(client mf.Client, m1, m2 *mf.Manifest) (*mf.Manifest, error) {
+	result, err := mf.ManifestFrom(mf.Slice(append(m1.Resources(), m2.Resources()...)))
+	if err != nil {
+		return nil, fmt.Errorf("failed to merge manifests: %w", err)
+	}
+	result.Client = client
```

Comment:
I'd avoid passing the client down in here and just set it on the manifest after creation.

Reply:
sounds like a good idea.